library/ecp_curves.c

```diff
@@ -836,7 +836,6 @@ int mbedtls_ecp_group_load( mbedtls_ecp_group *grp, mbedtls_ecp_group_id id )
 #endif /* MBEDTLS_ECP_DP_CURVE448_ENABLED */
 
         default:
-            mbedtls_ecp_group_free( grp );
```

Comment:
This leaves `grp->id` set to the invalid value `id`. Code that's using this function should stop using `grp` if the function returns a failure status, but it's more robust if `grp` stays in a valid state, with `id=0`.

Reply:
@gilles-peskine-arm thank you for the review! Fixed.